# Machine Learning and RNA
Lists of models and tools based on Machine Learning to solve problems related to RNA structure prediction, representation, classification, and many more

## Deep learning for predicting RNA secondary structures
- [Sequence similarity governs generalizability of de novo deep learning models for RNA secondary structure prediction](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1011047): "...Model generalizability, i.e., the performance gap between the seen and unseen sets, degrades rapidly as the sequence similarity decreases....  Generalizability thus poses a major hurdle for deploying de novo DL models in practice and various pathways for future advances are discussed."
- [Automatic recognition of complementary strands: Lessons regarding machine learning abilities in RNA folding](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1011047): "...We observed that low-capacity models are better suited for learning with mislabelled training examples, while large capacities improve the ability to generalize to structurally dissimilar data. Given a more complex task like RNA folding, it comes as no surprise that the scarcity of usable examples hurdles the applicability of machine learning techniques to this field."
- [Machine learning for RNA 2D structure prediction benchmarked on experimental data](https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbad153/7140288):"...We show that DL-based algorithms (such as SPOT-RNA and UFold) can outperform SL and traditional methods if the data distribution is similar in the training and testing set. However, when predicting 2D structures for new RNA families, the advantage of DL is no longer clear, and its performance is inferior or equal to that of SL and non-ML methods."

## RNA representation
- [Informative RNA base embedding for RNA structural alignment and clustering by deep representation learning](https://academic.oup.com/nargab/article/4/1/lqac012/6534363): '...we adopt a pre-training algorithm for the effective embedding of RNA bases to acquire semantically rich representations... '
- [An efficient graph kernel method for non-coding RNA functional prediction](https://academic.oup.com/bioinformatics/article/33/17/2642/3798629): '...We employ a flexible graph encoding to preserve multiple structural hypotheses and exploit recent advances in representation and model induction to scale to large data volumes...'
- [The super-n-motifs model: a novel alignment-free approach for representing and comparing RNA secondary structures](https://academic.oup.com/bioinformatics/article/33/8/1169/2907822): '...we propose the super-n-motifs model based on a latent analysis of enhanced motifs comprising not only basic motifs but also adjacency relations. The super-n-motifs model computes a vector representation of secondary structures as linear combinations of these motifs. ...'

## Tools 
- [e-RNA: a collection of web-servers for the prediction and visualisation of RNA secondary structure and their functional features](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkad296/7143234): e-RNA is a collection of web-servers for the prediction and visualisation of RNA secondary structures and their functional features, including in particular RNAâ€“RNA interactions
